tests/simulationTest.m

```diff
@@ -0,0 +1,108 @@
+classdef simulationTest < matlab.unittest.TestCase
+    %Test for simulation
+    properties 
+        Config;
+        Simulation;
+    end
+    methods (TestClassSetup)
+        function createObjects(testCase)
+            testCase.Config = MonsterConfig();
+
+            testCase.Simulation = Monster(testCase.Config);
+        end
+    end
+
+    methods (TestMethodTeardown)
+        function resetObjects(testCase)
+            %Test Teardown
+
+        end
+    end
+
+    methods (Test)
+        function testConstructor(testCase)
+            testCase.verifyTrue(isa(testCase.Simulation, 'Monster'));
+        end
+
+        function testSetupRound(testCase)
+            %Setup simulation for round 0
+            iRound = 0;
+            testCase.Simulation.setupRound(iRound);
+            testCase.verifyTrue(testCase.Config.Runtime.currentRound == iRound);
+            testCase.verifyTrue(testCase.Config.Runtime.currentTime == iRound*10e-3);
+            testCase.verifyTrue(testCase.Config.Runtime.remainingTime == (testCase.Config.Runtime.totalRounds - testCase.Config.Runtime.currentRound)*10e-3);
+            testCase.verifyTrue(testCase.Config.Runtime.remainingRounds == testCase.Config.Runtime.totalRounds - testCase.Config.Runtime.currentRound - 1);
+
+            %Test for channel setup as well???
+
+        end
+
+        function testRun(testCase)
+            testCase.Simulation.run();
+
+
+        end
+
+        function testCollectResults(testCase)
+            testCase.Simulation.collectResults();
+
+        end
+
+        function testClean(testCase)
+            testCase.Simulation.clean();
+            %Test that stations and users are reset
+            %Test Stations
+            arrayfun(@(x) testCase.verifyEqual(x.NSubframe, mod(1,10)) , testCase.Simulation.Stations);
+            %TODO: add picostations as well, they are currently disabled
```

Comment:
Create an issue on this. I think pico stations should be considered future work for now.